tests/unit/test_subscription_request.py

```diff
@@ -0,0 +1,272 @@
+        end_time=datetime(2023, 11, 1),
+        rrule="FREQ=MONTHLY;BYMONTH=3,4,5,6,7,8,9,10",
+    )
+
+    expected = {
+        "type": "catalog",
+        "parameters": {
+            "geometry": geom_geojson,
+            "start_time": "2021-03-01T00:00:00Z",
+            "end_time": "2023-11-01T00:00:00Z",
+            "rrule": "FREQ=MONTHLY;BYMONTH=3,4,5,6,7,8,9,10",
+            "item_types": ["PSScene"],
+            "asset_types": ["ortho_analytic_4b"]
+        }
+    }
+
+    assert res == expected
+
+
+def test_catalog_source_invalid_start_time(geom_geojson):
+    with pytest.raises(exceptions.ClientError):
+        subscription_request.catalog_source(
+            item_types=["PSScene"],
+            asset_types=["ortho_analytic_4b"],
+            geometry=geom_geojson,
+            start_time='invalid',
+            end_time=datetime(2023, 11, 1),
+            rrule="FREQ=MONTHLY;BYMONTH=3,4,5,6,7,8,9,10",
+        )
+
+
+def test_amazon_s3_success():
+    res = subscription_request.amazon_s3(aws_access_key_id='keyid',
+                                         aws_secret_access_key='accesskey',
+                                         bucket='bucket',
+                                         aws_region='region')
+
+    assert res == {
+        "type": "amazon_s3",
+        "parameters": {
+            "aws_access_key_id": "keyid",
+            "aws_secret_access_key": "accesskey",
+            "bucket": "bucket",
+            "aws_region": "region"
+        }
+    }
+
+
+def test_azure_blob_storage_success():
+    res = subscription_request.azure_blob_storage(
+        account='act',
+        container='container',
+        sas_token='sastoken',
+        storage_endpoint_suffix='suffix')
+
+    assert res == {
+        "type": "azure_blob_storage",
+        "parameters": {
+            "account": "act",
+            "container": "container",
+            "sas_token": "sastoken",
+            "storage_endpoint_suffix": "suffix"
+        }
+    }
+
+
+def test_google_cloud_storage_success():
+    res = subscription_request.google_cloud_storage(credentials='cred',
+                                                    bucket='bucket')
+
+    assert res == {
+        "type": "google_cloud_storage",
+        "parameters": {
+            "bucket": "bucket", "credentials": "cred"
+        }
+    }
+
+
+def test_oracle_cloud_storage_success():
+    res = subscription_request.oracle_cloud_storage(
+        customer_access_key_id='keyid',
+        customer_secret_key='secretkey',
+        bucket='bucket',
+        region='region',
+        namespace='namespace')
+
+    assert res == {
+        "type": "oracle_cloud_storage",
+        "parameters": {
+            "customer_access_key_id": "keyid",
+            "customer_secret_key": "secretkey",
+            "bucket": "bucket",
+            "region": "region",
+            "namespace": "namespace"
+        }
+    }
+
+
+def test_notifications_success():
```

Comment:
Would you consider it overkill to test for each notification topic? For instance:
```
@pytest.mark.parametrize("topics", subscription_request.NOTIFICATIONS_TOPICS)
def test_notifications_success(topics):
   notifications_config = subscription_request.notifications(url='url',
                                                             topics=topics)

    expected = {"webhook": {"url": "url", "topics": topics}}
    assert notifications_config == expected
```